### PR TITLE
Implements Realm routes

### LINKED ISSUE #1610: Create realm routes

### PR DESCRIPTION
**Pull request by imsdu**
Fixes #1610

### CODE REVIEW

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 1]
```diff
@@ -102,13 +133,16 @@ trait DeltaDirectives extends RdfMarshalling {
   def requestMediaType: Directive1[MediaType] =
     extractRequest.flatMap { req =>
       val ct = new MediaTypeNegotiator(req.headers)
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:135]: It is simpler to keep the logic and just add the */* mediaRange to the acceptedMediaRanges to the list, as follow:

```scala
(ct.acceptedMediaRanges :+ `*/*`).foldLeft[Option[MediaType]](None) { ... }
```

Since the RFC2616 says
> If no Accept header field is present, then it is assumed that the client accepts all media types

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 2]
```diff
@@ -174,135 +206,241 @@ trait DeltaDirectives extends RdfMarshalling {
...
       io: UIO[A]
   )(implicit s: Scheduler, cr: RemoteContextResolution, ordering: JsonKeyOrdering): Route =
-    completeUIO(status, Seq.empty, io)
+    completeUIOOpt(status, io.map(Some(_)))
 
   /**
     * Completes a passed [[UIO]] of ''A'' with the desired output format using the implicitly available [[JsonLdEncoder]].
+    * If the normal channel doesn't hold any value, a not found output is produced
+    *
+   * @param status the returned HTTP status code
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:215]: wrong formatting

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 3]
```diff
@@ -174,135 +206,241 @@ trait DeltaDirectives extends RdfMarshalling {
...
+          case Result(status, headers, dot) =>
+            complete(status, headers, dot)
         }
     }
 
   /**
     * Completes a passed [[IO]] of ''E'' and ''A'' with the desired output format using the implicitly available [[JsonLdEncoder]].
     * Both error channel and normal channel are converted to the desired output format.
     *
+   * @param status  the returned HTTP status code
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:369]: wrong formatting

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 4]
```diff
@@ -174,135 +206,241 @@ trait DeltaDirectives extends RdfMarshalling {
...
       ordering: JsonKeyOrdering
   ): Route =
-    completeIO(StatusCodes.OK, Seq.empty, io)
+    completeIOOpt(StatusCodes.OK, Seq.empty, io)
+
+  /**
+    * Completes a passed [[IO]] of ''E'' and ''A'' with the desired output format using the implicitly available [[JsonLdEncoder]].
+    * Both error channel and normal channel are converted to the desired output format.
+    *
+   * @param io  the value to be returned, wrapped in an [[IO]]
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:435]: wrong formatting

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 5]
```diff
@@ -330,4 +468,42 @@ trait DeltaDirectives extends RdfMarshalling {
...
-object DeltaDirectives extends DeltaDirectives
+object DeltaDirectives {
+
+  private val notFound: ServiceError = ServiceError.NotFound
+
+  private[routes] case class Result[C](statusCode: StatusCode, hearders: Seq[HttpHeader], content: C)
+
+  object Result {
+
+    def compactedJsonLd[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:479]: docs?

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 6]
```diff
@@ -330,4 +468,42 @@ trait DeltaDirectives extends RdfMarshalling {
...
+        implicit cr: RemoteContextResolution
+    ): IO[RdfError, Result[JsonLd]] =
+      value.fold(notFound.toCompactedJsonLd.map(v => Result(StatusCodes.NotFound, Nil, v: JsonLd))) { r: A =>
+        r.toCompactedJsonLd.map { v => Result(successStatus, successHeaders, v) }
+      }
+
+    def expandedJsonLd[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:486]: docs?

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 7]
```diff
@@ -330,4 +468,42 @@ trait DeltaDirectives extends RdfMarshalling {
...
+      }
+
+    def nTriples[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(
+        implicit cr: RemoteContextResolution
+    ): IO[RdfError, Result[NTriples]] =
+      value.fold(notFound.toNTriples.map(v => Result(StatusCodes.NotFound, Nil, v))) { r: A =>
+        r.toNTriples.map { v => Result(successStatus, successHeaders, v) }
+      }
+
+    def dot[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(implicit
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:500]: docs?

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala [hunk 8]
```diff
@@ -330,4 +468,42 @@ trait DeltaDirectives extends RdfMarshalling {
...
+      }
+
+    def expandedJsonLd[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(
+        implicit cr: RemoteContextResolution
+    ): IO[RdfError, Result[JsonLd]] =
+      value.fold(notFound.toExpandedJsonLd.map(v => Result(StatusCodes.NotFound, Nil, v: JsonLd))) { r: A =>
+        r.toExpandedJsonLd.map { v => Result(successStatus, successHeaders, v) }
+      }
+
+    def nTriples[A: JsonLdEncoder](value: Option[A], successStatus: StatusCode, successHeaders: Seq[HttpHeader])(
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:493]: docs?

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractSubject { implicit subject =>
+        pathPrefix("realms") {
+          concat(
+            (get & extractUri & paginated & searchParams & pathEndOrSingleSlash) { (uri, pagination, params) =>
+              operationName(s"$prefixSegment/realms") {
+                implicit val searchResultEncoder: Encoder.AsObject[SearchResults[RealmResource]] =
```
Comment (umbreak) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:64]: This is quite noisy. I think we could have the jsonldEncoder[SearchResults[RealmResource]] be generated implicitly from a `Pagination` + `Uri` (or something wrapped on a Uri like `RequestUri`)

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceF.scala [hunk 10]
```diff
@@ -78,6 +88,14 @@ object ResourceF {
       }
     }
 
+  implicit def resourceFAEncoder[A](implicit
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/ResourceF.scala:91]: More concise:
```scala
implicit def resourceFAEncoder[A: Encoder.AsObject](implicit base: BaseUri): Encoder.AsObject[ResourceF[Iri, A]] =
Encoder.AsObject.instance ( r => r.void.asJsonObject deepMerge r.value.asJsonObject)
```

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala (hunk 9 above)
Comment (imsdu) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:64]: Yep, I am not that sure about that part, I will try to make it better.
I tried to make a version where all is generated implicitly and it was quite a implicit resolution mess so that's why I tried this

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala (hunk 9 above)
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:64]: We can have a directive called `extractRequestUri ` which returns a `Directive1[RequestUri]` and so we can safely use the implicit `Pagination` and `RequestUri` fields to generate the jsonLdEncoder needed.

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala (hunk 1 above)
Comment (imsdu) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:135]: Won't it break the compatibility for clients ?

**Comment by umbreak** [comment]
Since you have added this `Metadata` to the `Vocabulary`, we should also change the fields that now are hardcoded elsewhere (I can see some "_rev", "_deprecated")

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala (hunk 1 above)
Comment (umbreak) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:135]: Well, it is what the RFC states, so I assume it is the right thing to do.

File: delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/Vocabulary.scala [hunk 11]
```diff
@@ -94,9 +95,32 @@ object Vocabulary {
...
+    val authorizationEndpoint = Metadata("authorizationEndpoint")
+    val createdAt             = Metadata("createdAt")
+    val createdBy             = Metadata("createdBy")
+    val deprecated            = Metadata("deprecated")
+    val endSessionEndpoint    = Metadata("endSessionEndpoint")
+    val grantTypes            = Metadata("grantTypes")
+    val issuer                = Metadata("issuer")
+    val maxScore              = Metadata("maxScore")
+    val next                  = Metadata("next")
+    val label                 = Metadata("label")
```
[the remark this answers is not in the document]
Comment (imsdu) [reply delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/Vocabulary.scala:111]: It was there in the previous implementation. Should I remove it then ?

File: delta/sdk/src/main/resources/contexts/realms.json [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    "logo": {
+      "@type": "@id",
+      "@id": "nxv:logo"
+    },
+    "openIdConfig": {
+      "@id": "nxv:openIdConfig",
+      "@type": "@id"
+    },
+    "name": "nxv:name",
+    "_label": "nxv:label",
```
[the remark this answers is not in the document]
Comment (imsdu) [reply delta/sdk/src/main/resources/contexts/realms.json:13]: As an id for realm yes. But I don't know if it should be an id or not. It is in 1.4

[the remark this answers is not in the document]
Comment (umbreak) [reply delta/sdk/src/main/resources/contexts/realms.json:13]: Keep it as metadata. It is provided as a path parameter but not as payload.

**Comment by imsdu** [comment]
@umbreak  About replacing _rev and _deprecated, I will take care of it in a second PR to avoid adding noise in this one

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 13]
```diff
@@ -12,8 +11,8 @@ import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceRef}
 sealed trait SearchParams {
   def deprecated: Option[Boolean]
   def rev: Option[Long]
-  def createdBy: Option[Subject]
-  def updatedBy: Option[Subject]
+  def createdBy: Option[Iri]
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: It is better to have `Subject` here, since in other parts of the code is better to deal with `Subject` than with `Iri`

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala [hunk 14]
```diff
@@ -12,8 +11,8 @@ import ch.epfl.bluebrain.nexus.delta.sdk.model.{Label, ResourceRef}
...
+  def updatedBy: Option[Iri]
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:15]: same

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 13 above)
Comment (imsdu) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: Yes but the problem is using a subject brings complexity during deserialization

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala (hunk 1 above)
Comment (bogdanromanx) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/DeltaDirectives.scala:135]: I think it's fine to follow the RFC... I can't see how this would break the compatibility.

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala (hunk 13 above)
Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: This issue in deserialization was already dealt with when implementing `Permissions`. Having an implicit BaseUri you can always extract the Iri from the Subject

Comment (imsdu) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: In Permissions, it must be another usage, here we extract the Iri from a get parameter to build the `SearchParams`.
To build a subject, we either have to add a discriminator in the get parameter or parse first as an Iri and then parse the Iri as a subject.

Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: I guess the directive which extracts the SearchParams should do that, right? And yes, we have to parse the Iri and from it generate an Identity.

In the Identity class we need an apply method as follows:
```scala
def apply(iri: Iri): Either[TokenRejection, Identity] = ???
```

Comment (imsdu) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: Is this refinement worth the amount of additional code and tests though ?

Comment (umbreak) [reply delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/search/SearchParams.scala:14]: I think we are going to need the conversion from `Iri => Identity` anyways (like we now have the one from Identity => Iri). So I would say yes.

But let's see what the others say.

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+        RealmSearchParams(None, deprecated, rev, createdBy, updatedBy)
+    }
+
+  def routes: Route =
+    baseUriPrefix(baseUri.prefix) {
+      extractSubject { implicit subject =>
+        pathPrefix("realms") {
+          concat(
+            (get & extractUri & paginated & searchParams & pathEndOrSingleSlash) { (uri, pagination, params) =>
+              operationName(s"$prefixSegment/realms") {
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:62]: the operationName should be prefixed with a `/`

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+  def searchParams: Directive1[RealmSearchParams] =
+    (parameter("deprecated".as[Boolean].?) &
+      parameter("rev".as[Long].?) &
+      parameter("createdBy".as[Iri].?) &
+      parameter("updatedBy".as[Iri].?)).tmap {
+      case (deprecated, rev, createdBy, updatedBy) =>
+        RealmSearchParams(None, deprecated, rev, createdBy, updatedBy)
+    }
+
+  def routes: Route =
```
Comment (bogdanromanx) [inline delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:56]: missing the event routes

File: delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutesSpec.scala [hunk 17]
```diff
@@ -0,0 +1,214 @@
...
+
+    "fail fetching a realm by id and rev when rev is invalid" in {
+      Get("/v1/realms/github?rev=4") ~> routes ~> check {
+        status shouldEqual StatusCodes.NotFound
+        response.asJson shouldEqual jsonContentOf("/realms/revision-not-found.json", "provided" -> 4, "current" -> 2)
+      }
+    }
+
+    "list realms" in {
+      Get("/v1/realms") ~> routes ~> check {
```
Comment (bogdanromanx) [inline delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutesSpec.scala:195]: I think we should also test the search params

File: delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutesSpec.scala [hunk 18]
```diff
@@ -0,0 +1,214 @@
...
+  )
+
+  private val gitlabDeprecated = gitlabCreated.deepMerge(
+    Json.obj(
+      "_deprecated" -> Json.fromBoolean(true),
+      "_rev"        -> Json.fromLong(2L)
+    )
+  )
+
+  "A RealmsRoute" should {
```
Comment (bogdanromanx) [inline delta/app/src/test/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutesSpec.scala:133]: I would also include a test that covers realm issuer collision response

**Review by umbreak — APPROVED** [review]

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala (hunk 15 above)
Comment (imsdu) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:62]: prefixSegment is prefixed with a / so it is more the others operationNames defined that are wrong

Comment (bogdanromanx) [reply delta/app/src/main/scala/ch/epfl/bluebrain/nexus/delta/routes/RealmsRoutes.scala:62]: the others should be corrected in this case :)

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Realms.scala [hunk 19]
```diff
@@ -90,6 +92,14 @@ trait Realms {
       params: RealmSearchParams = RealmSearchParams.none
   ): UIO[UnscoredSearchResults[RealmResource]]
 
+  /**
+    * A non terminating stream of events for permissions. After emitting all known events it sleeps until new events
+    * are recorded.
+    *
+   * @param offset the last seen event offset; it will not be emitted by the stream
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Realms.scala:99]: formatting

File: delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/realms/RealmEvent.scala [hunk 20]
```diff
@@ -108,4 +117,46 @@ object RealmEvent {
...
       subject: Subject
   ) extends RealmEvent
+
+  import GrantType.Camel._
+  import ch.epfl.bluebrain.nexus.delta.sdk.instances._
+
+  private val context = ContextValue(contexts.resource, contexts.realms)
+
+  implicit private[realms] val config: Configuration = Configuration.default
+    .withDiscriminator("@type")
```
Comment (umbreak) [inline delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/realms/RealmEvent.scala:127]:
```scala
.withDiscriminator(keywords.tpe)
```

File: delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala [hunk 21]
```diff
@@ -30,10 +34,21 @@ final class RealmsDummy private (
...
     cache: IORef[RealmsCache],
-    semaphore: IOSemaphore
+    semaphore: IOSemaphore,
+    maxStreamSize: Long
 )(implicit clock: Clock[UIO], base: BaseUri)
     extends Realms {
 
+  private val offsetMax = new AtomicLong()
+
+  private def journalMap: IO[Nothing, Map[Label, Vector[Envelope[RealmEvent]]]] =
```
Comment (umbreak) [inline delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala:44]: `Map[Label, RealmsJournal]`

File: delta/sdk-testkit/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsBehaviors.scala [hunk 22]
```diff
@@ -188,6 +191,40 @@ trait RealmsBehaviors {
       realms.deprecate(github, 3L).rejectedWith[RealmAlreadyDeprecated]
     }
 
+    val quote     = Pattern.quote("$")
+    val allEvents = List(
+      (github, RealmCreated.getClass.getSimpleName.replaceAll(quote, ""), Sequence(1L)),
```
Comment (umbreak) [inline delta/sdk-testkit/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsBehaviors.scala:196]: Seems that we are using this method to get the name of a class in many places.
Another variation was:
```scala
value.getClass.getSimpleName.split('$').head
```

In any case, shouldn't we put that somewhere in the SDK as some utils? (it is use in all the rejections for example).

However we can do that in a separate PR if you want.

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala [hunk 23]
```diff
@@ -108,30 +135,44 @@ final class RealmsDummy private (
...
         res      <- IO.fromEither(Realms.next(state, event).toResource.toRight(UnexpectedInitialState(cmd.label)))
       } yield res
     }
+
+  private def makeEnvelope(event: RealmEvent): Envelope[RealmEvent] = {
+    Envelope(
+      event,
+      event.getClass.getSimpleName,
+      Sequence(offsetMax.incrementAndGet()),
+      s"realms-${event.label}",
```
Comment (umbreak) [inline delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala:150]:
```scala
s"$entityType-${event.label}"
```

File: delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala [hunk 24]
```diff
@@ -78,27 +93,39 @@ final class RealmsDummy private (
...
     cache.update(_.updatedWith(resource.id)(_.as(resource))).as(resource)
 
   private def currentState(label: Label): UIO[Option[RealmState]] =
-    journal.get.map { labelsEvents =>
-      labelsEvents.get(label).map(_.foldLeft[RealmState](Initial)(Realms.next))
+    journalMap.map { labelsEvents =>
+      labelsEvents
+        .get(label)
+        .map(_.foldLeft[RealmState](Initial) {
+          case (s, e) =>
```
Comment (umbreak) [inline delta/sdk-testkit/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsDummy.scala:114]: you don't need `case` here.
```scala
_.foldLeft[RealmState](Initial)((state, eventEnvelope) => Realms.next(state, eventEnvelope.event))
```

**Review by umbreak — CHANGES_REQUESTED** [review]

File: delta/sdk-testkit/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsBehaviors.scala (hunk 22 above)
Comment (imsdu) [reply delta/sdk-testkit/src/test/scala/ch/epfl/bluebrain/nexus/delta/sdk/testkit/RealmsBehaviors.scala:196]: Yep, we can do it in another PR to avoid more noise

**Review by umbreak — DISMISSED** [review]

**Review by umbreak — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]
